### PR TITLE
Add tests for `base64URLEncode` and `base64URLDecode`. Add analyzer tests

### LINKED ISSUE #56337: A list of unusual queries that only work with Analyzer
We will add them to functional tests.

```
SELECT *, ngramMinHash(*) AS minhash, mortonEncode(untuple(ngramMinHash(*))) AS z FROM symbols ORDER BY z LIMIT 100
```

```
WITH
    4096 AS w, 4096 AS h, w * h AS pixels,
    arrayJoin(coverage) AS num,
    num DIV (32768 * 32768 DIV pixels) AS idx,
    mortonDecode(2, idx) AS coord,
    255 AS b,
    least(255, uniq(test_name)) AS r,
    255 * uniq(test_name) / (max(uniq(test_name)) OVER ()) AS g
SELECT r::UInt8, g::UInt8, b::UInt8
FROM system.coverage_
GROUP BY coord
ORDER BY coord.2 * w + coord.1
WITH FILL FROM 0 TO 4096*2048
```

### PR DESCRIPTION
**Pull request by fm4v**
Closes: https://github.com/ClickHouse/ClickHouse/issues/56337

### Changelog category (leave one):
- Build/Testing/Packaging Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Add tests for `base64URLEncode` and `base64URLDecode`. Add analyzer tests

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ]  Allow: All Required Checks
- [ ]  Allow: Stateless tests
- [ ]  Allow: Stateful tests
- [ ]  Allow: Integration Tests
- [ ]  Allow: Performance tests
- [ ]  Allow: All Builds
- [ ]  Allow: batch 1, 2 for multi-batch jobs
- [ ]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [ ]  Exclude: Fast test
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
- [ ]  Exclude: All with aarch64, release, debug
---
- [ ]  Do not test
- [ ]  Woolen Wolfdog
- [ ]  Upload binaries for special builds
- [ ]  Disable merge-commit
- [ ]  Disable CI cache

### CODE REVIEW

File: tests/queries/0_stateless/03167_base64_url_functions_sh.sh [hunk 1]
```diff
@@ -0,0 +1,191 @@
+#!/usr/bin/env bash
+# Tags: no-fastest, no-debug
```
Comment (antaljanosbenjamin) [inline tests/queries/0_stateless/03167_base64_url_functions_sh.sh:2]: Why no fastest and debug?

**Review by antaljanosbenjamin — APPROVED** [review]

**Comment by robot-ch-test-poll3** [comment]
*This is an automated comment for commit c4da270ddb67e0710e35e0ed022b91280d30dc70 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/65979/c4da270ddb67e0710e35e0ed022b91280d30dc70/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Builds</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/138eb924fa31351f2998651ea17eca8f7264f481/builds/report.html">✅ success</a></td></tr>
<tr><td>ClickBench</td><td>Runs [ClickBench](https://github.com/ClickHouse/ClickBench/) with instant-attach table</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/138eb924fa31351f2998651ea17eca8f7264f481/clickbench__aarch64_.html">✅ success</a></td></tr>
<tr><td>Compatibility check</td><td>Checks that <b>clickhouse</b> binary runs on distributions with old libc versions. If it fails, ask a maintainer for help</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/138eb924fa31351f2998651ea17eca8f7264f481/compatibility_check__aarch64_.html">✅ success</a></td></tr>
<tr><td>Docker keeper image</td><td>The check to build and optionally push the mentioned image to docker hub</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/138eb924fa31351f2998651ea17eca8f7264f481/docker_keeper_image.html">✅ success</a></td></tr>
<tr><td>Docker server image</td><td>The check to build and optionally push the mentioned image to docker hub</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/138eb924fa31351f2998651ea17eca8f7264f481/docker_server_image.html">✅ success</a></td></tr>
<tr><td>Docs check</td><td>Builds and tests the documentation</td><td>✅ success</td></tr>
<tr><td>Install packages</td><td>Checks that the built packages are installable in a clear environment</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/138eb924fa31351f2998651ea17eca8f7264f481/install_packages__aarch64_.html">✅ success</a></td></tr>
<tr><td>Integration tests</td><td>The integration tests report. In parenthesis the package type is given, and in square brackets are the optional part/total tests</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/138eb924fa31351f2998651ea17eca8f7264f481/integration_tests__aarch64__[1_6].html">✅ success</a></td></tr>
<tr><td>Performance Comparison</td><td>Measure changes in query performance. The performance test report is described in detail <a href="https://github.com/ClickHouse/ClickHouse/tree/master/docker/test/performance-comparison#how-to-read-the-report">here</a>. In square brackets are the optional part/total tests</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/138eb924fa31351f2998651ea17eca8f7264f481/performance_comparison__release__[3_4]/report.html">✅ success</a></td></tr>
<tr><td>Stateful tests</td><td>Runs stateful functional tests for ClickHouse binaries built in various configurations -- release, debug, with sanitizers, etc</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/138eb924fa31351f2998651ea17eca8f7264f481/stateful_tests__aarch64_.html">✅ success</a></td></tr>
<tr><td>Unit tests</td><td>Runs the unit tests for different release types</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/138eb924fa31351f2998651ea17eca8f7264f481/unit_tests__asan_.html">✅ success</a></td></tr>
<tbody>
</table>
</details>

**Comment by fm4v** [comment]
@antaljanosbenjamin i've added more tests unrelated to PR name, going to update description

File: tests/queries/0_stateless/03167_base64_url_functions_sh.sh [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+    "http:///a"
+    "http://example.com??"
+    "http://example.com??/"
+    "foo://example.com:8042/over/there?name=ferret#nose"
+    "//example.com/path"
+)
+
+
+base64URLEncode() {
+    echo -n "$1" | openssl base64 -e -A | tr '+/' '-_' | tr -d '='
```
Comment (rschu1ze) [inline tests/queries/0_stateless/03167_base64_url_functions_sh.sh:122]: `openssl` seem overkill, linux comes with the `base64` CLI tool.

File: tests/queries/0_stateless/03199_queries_with_new_analyzer.sql [hunk 3]
```diff
@@ -0,0 +1,41 @@
+SET allow_experimental_analyzer=1;
```
Comment (rschu1ze) [inline tests/queries/0_stateless/03199_queries_with_new_analyzer.sql:1]: Not related to base64 tests, no?

File: tests/queries/0_stateless/03167_base64_url_functions_sh.sh [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+# shellcheck disable=SC2155
+
+set -e
+
+CURDIR=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
+# shellcheck source=../shell_config.sh
+. "$CURDIR"/../shell_config.sh
+
+
+urls=(
```
Comment (rschu1ze) [inline tests/queries/0_stateless/03167_base64_url_functions_sh.sh:11]: There are already some tests for base64URL in tests/queries/0_stateless/03167_base64_url_functions.sql

Consider merging these into this file (only if that makes sense) to keep things together.

File: tests/queries/0_stateless/03167_base64_url_functions_sh.sh (hunk 2 above)
Comment (fm4v) [reply tests/queries/0_stateless/03167_base64_url_functions_sh.sh]: Yes, it's better

File: tests/queries/0_stateless/03199_queries_with_new_analyzer.sql (hunk 3 above)
Comment (fm4v) [reply tests/queries/0_stateless/03199_queries_with_new_analyzer.sql:1]: No

File: tests/queries/0_stateless/03167_base64_url_functions_sh.sh (hunk 4 above)
Comment (fm4v) [reply tests/queries/0_stateless/03167_base64_url_functions_sh.sh:11]: It's ok to have multiple tests on single feature)

**Comment by fm4v** [comment]
Fasttest.......
```
DB::Exception: Function with name 'base64URLEncode' does not exist. In scope SELECT base64URLEncode('http://www.example.com'). Maybe you meant: ['base58Encode']. (UNKNOWN_FUNCTION) (version 24.7.1.1) (from [::ffff:127.0.0.1]:46462) (comment: 03167_base64_url_functions_sh.sh)
```

**Comment by rschu1ze** [comment]
@fm4v This is expected, FastTest runs on top of a ClickHouse build with a minimum number of 3rd party libraries (to ensure fast compilation). Base64 encoding/decoding uses a library which is not available in this build, as a result no related SQL functions can be used.

Add this to the first line of the test: `# Tags: no-fasttest`

**Pull request merged**